dev/RadioMenuFlyoutItem/RadioMenuFlyoutItem.cpp

```diff
@@ -7,15 +7,32 @@
 #include "RuntimeProfiler.h"
 #include "ResourceAccessor.h"
 
+winrt::IMap<winrt::hstring, winrt::RadioMenuFlyoutItem> RadioMenuFlyoutItem::s_selectionMap = nullptr;
+
 RadioMenuFlyoutItem::RadioMenuFlyoutItem()
 {
     __RP_Marker_ClassById(RuntimeProfiler::ProfId_RadioMenuFlyoutItem);
 
     m_InternalIsCheckedChangedRevoker = RegisterPropertyChanged(*this, winrt::ToggleMenuFlyoutItem::IsCheckedProperty(), { this, &RadioMenuFlyoutItem::OnInternalIsCheckedChanged });
 
+    if (!s_selectionMap)
+    {
+        // Ensure that this object exists
+        s_selectionMap = winrt::single_threaded_map<winrt::hstring, winrt::RadioMenuFlyoutItem>();
```

Comment:
>winrt::RadioMenuFlyoutItem [](http://example.com/codeflow?start=68&length=26)

This is your problem. A strong reference here is why your RadioMenuFlyoutItem never gets cleaned up. Make this winrt::weak_ref<winrt::RadioMenuFlyoutItem>

Reply:
ah HA